src/utils/task-queue.js

```diff
@@ -53,14 +63,24 @@ export default options => {
   const queue = async.queue((task, cb) => {
     auth.shouldRetrieveToken(shouldRetrieve => {
       if (shouldRetrieve)
-        if (auth.accessToken) execTask(task, cb)
+        if (hasValidAccessToken()) execTask(task, cb)
         else {
           queue.pause()
 
-          authUtils.getAccessToken(options)
+          authUtils.getAccessToken(Object.assign({}, options, {
+            auth: Object.assign({}, options.auth, {
+              accessToken: undefined,
+              accessTokenExpirationTime: undefined,
+            }),
```

Comment:
If I don't set `accessToken` to `undefined` here `getAccessToken` will sent the expired token in the `Authorization` header to server which results in an error, because the server thinks I want to retrieve a new token by sending the client credentials in the `Authorization` header instead of sending them in the url (http://dev.commercetools.com/http-api-authorization.html#client-credentials-flow).